### PR TITLE
Allow empty response

### PR DESCRIPTION
**Pull request by stan-is-hate**
This PR allows empty message responses. One use case is grpc error responses, where you might want to ensure the server responded with no message but just the status.

TODOs:
- make sure CI works (avro plugin didn't start correctly on my machine, but might be that it doesn't support arm macs)
- ~do we want to allow empty responses only when metadata is not empty? and if so, how to do it?~ I think this should be handled in the rust part, not here. Here all we do is decode and pass the message into the test func.
- ~why aren't matching rules supported in the metadata (might need a separate PR)~ https://github.com/pact-foundation/pact-reference/pull/384 
- ~can this be tested via unit tests~ done, added unit tests
- squash commits once reviewed (don't want to squash because changes might be required, wouldn't want to squash again)

### CODE REVIEW

File: examples/provider_test.go [hunk 1]
```diff
@@ -51,13 +51,14 @@ func TestV3HTTPProvider(t *testing.T) {
...
 		ProviderBaseURL: "http://127.0.0.1:8111",
-		Provider:        "V3Provider",
+		Provider:        "V4Provider",
 		ProviderVersion: os.Getenv("APP_SHA"),
-		BrokerURL:       os.Getenv("PACT_BROKER_BASE_URL"),
-		PactFiles: []string{
-			filepath.ToSlash(fmt.Sprintf("%s/PactGoV3Consumer-V3Provider.json", pactDir)),
-			filepath.ToSlash(fmt.Sprintf("%s/PactGoV2ConsumerMatch-V2ProviderMatch.json", pactDir)),
-		},
+		// BrokerURL:       os.Getenv("PACT_BROKER_BASE_URL"),
```
Comment (mefellows) [inline examples/provider_test.go:56]: Did you mean to leave these uncommented?

**Review by mefellows — APPROVED** [review]
Couple of minor changes. Thanks so much for this 🙏

File: examples/grpc/grpc_consumer_test.go [hunk 2]
```diff
@@ -93,3 +92,144 @@ func TestGrpcInteraction(t *testing.T) {
...
+			}
+
+			// Now we can make a normal gRPC request
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			_, err = c.GetFeature(ctx, point)
+
+			require.Error(t, err)
+			// TODO: uncomment once fixed on the ffi side
+			// require.ErrorContains(t, err, "no feature was found at latitude:-1  longitude:-1")
```
[the remark this answers is not in the document]
Comment (stan-is-hate) [reply examples/grpc/grpc_consumer_test.go:152]: We're dependent on this one instead: https://github.com/pact-foundation/pact-reference/commit/29b326e59b48a6a78a019b37e378b7742c728da5
my fix is for rust tests, Ronald fixed it for FFI

File: examples/provider_test.go (hunk 1 above)
Comment (stan-is-hate) [reply examples/provider_test.go:56]: Yeah, I think these changes crept in from various manual testing, I'll go through and clean them up tomorrow.

**Comment by stan-is-hate** [comment]
I'll close this one if you don't mind - I messed up the rebases and it's more hassle than it's worth to untangle them. I've opened #386 using a patch instead, so that one has a single commit :)

**Pull request closed without merging**